src/GrpAb/Elem.jl

```diff
@@ -224,6 +224,65 @@ end
 
 *(x::FinGenAbGroupElem, y::Integer) = y*x
 
+###############################################################################
+#
+#   Unsafe operators
+#
+###############################################################################
+
+function zero!(x::FinGenAbGroupElem)
+  zero!(x.coeff)
+  # no reduction necessary
+  return x
+end
+
+function neg!(x::FinGenAbGroupElem)
+  neg!(x.coeff)
+  assure_reduced!(parent(x), x.coeff)
+  return x
+end
+
+# TODO: set! for ZZMatrix not yet implemented, hence this is not yet implemented
+#function set!(x::FinGenAbGroupElem, y::FinGenAbGroupElem)
+#  set!(x.coeff, y.coeff)
+#  # no reduction necessary
+#  return x
+#end
+
+function add!(x::FinGenAbGroupElem, y::FinGenAbGroupElem, z::FinGenAbGroupElem)
+  add!(x.coeff, y.coeff, z.coeff)
+  assure_reduced!(parent(x), x.coeff)
+  return x
```

Comment:
```suggestion
  return reduce!(x)
```
I think I would find this even easier to understand (and less code duplication). Same for the other functions. But this is just personal preference

Reply:
Yeah makes sense. (This patch grew organically and `reduced!` was only added at the very end :-))